app/src/lib/menu-item.ts

```diff
@@ -0,0 +1,40 @@
+export interface IMenuItem {
+  /** The user-facing label. */
+  readonly label?: string
+
+  /** The action to invoke when the user selects the item. */
+  readonly action?: () => void
+
+  /** The type of item. */
+  readonly type?: 'separator'
+
+  /** Is the menu item enabled? Defaults to true. */
+  readonly enabled?: boolean
+
+  /**
+   * The predefined behavior of the menu item.
+   *
+   * When specified the click property will be ignored
+   */
+  readonly role?:
+    | 'undo'
```

Comment:
Can we extract this out into a separate type?

Reply:
I spent some time thinking about this and I decided to just remove the type here. That list of roles wasn't even complete and I think we should wait until Electron adds types for its role property and then reuse that instead of trying to stay in sync with what Electron does